caikit_nlp/toolkit/text_generation/model_run_utils.py

```diff
@@ -251,14 +266,37 @@ def generate_text_func(
 
     return GeneratedTextResult(
         generated_tokens=token_count,
-        generated_text=preds[0],
+        generated_text=generated_text,
         finish_reason=finish_reason,
         producer_id=producer_id,
         input_token_count=input_token_count,
         seed=seed,
     )
 
 
+def __postprocess_remove_input_text(tokenizer, preds, inputs, task_type):
+    """For Causal LM task types, preserve_input_text set to False will
+    remove the input text from generated output.
+    """
+    if task_type == "CAUSAL_LM":
```

Comment:
HF pipeline has implemented this for Causal LMs generation pipelines https://github.com/huggingface/transformers/blob/main/src/transformers/pipelines/text_generation.py#L287 

I took the same logic as it was suggested by other stack overflow posts too. I set the flag to True by default as generate() returns full text by default. 

I have tested this with fine tuning on llama model with different texts and found it to work as expected. Unit tests show the same